enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+
+Ingress Operator will create an NLB Ingress Controller with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+populated with the value from the eip-allocations field.
+
+#### Cluster administrator updates eip-allocations field the existing ingress controller
+
+Ingress Operator will update an NLB Ingress Controller with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+populated with the value from the eip-allocations field.
+
+### API Extensions
+- The API for setting `eip-allocations` field in the installer-config will look as follows in the file `pkg/types/aws/platform.go`:
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+...
+    // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/load-balancer-types.html#nlb 
+    // +optional 
+    LBType configv1.AWSLBType `json:"lbType,omitempty"`
+
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *configv1.AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+}
+
+```
+
+- The API for the `eip-allocations` field in the ingress cluster object to store the value of the
+  field `eip-allocations` from the installer which the ingress operator uses to create the ingress controller
+  will look as follows in the file `config/v1/types_ingress.go`:
+
+```go
+// AWSIngressSpec holds the desired state of the Ingress for Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+// +union
+type AWSIngressSpec struct {
+...
+    // +unionDiscriminator
+    // +kubebuilder:validation:Enum:=NLB;Classic
+    // +kubebuilder:validation:Required
+    Type AWSLBType `json:"type,omitempty"`
+
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+}
+
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer.
+type AWSNetworkLoadBalancerParameters struct {
+    EIPAllocations EIPAllocations `json:"eip-allocations,omitempty"`
+}
+
+type EIPAllocations string
+
+```
+
+- The API field for `eip-allocations` field in the Ingress Controller CR needs to be set as follows
+  in the file `operator/v1/types_ingress.go`:
+```go
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer.
+type AWSNetworkLoadBalancerParameters struct {
+    EIPAllocations EIPAllocations `json:"eip-allocations,omitempty"`        
+}
+
+type EIPAllocations string
+
+```
```

Comment:
We can do something like this
// AWSNetworkLoadBalancerParameters holds configuration parameters for an
// AWS Network load balancer.
type AWSNetworkLoadBalancerParameters struct {
	EIPAllocations configv1.EIPAllocations `json:"eip-allocations,omitempty"`
}

Reply:
I believe this info isn't needed in operator/v1/types_ingress.go

Reply:
not sure what you mean please check https://github.com/openshift/api/pull/1826/commits/261093f2b1ea2e427f113bcc4f4b0d3c1aca6d5a#diff-891d2cdcb01603ef2a2c1bfb2f6ef32cf2acf8b1471871a5a6e27308d50050f9R641

Reply:
Okay, my confusion was because the struct was empty to begin with.  I thought you were adding a new struct type.  Btw, your api PR shows this as a slice:

```
type AWSNetworkLoadBalancerParameters struct {
	EIPAllocations []configv1.EIPAllocations `json:"eipAllocations,omitempty"`
}
```

Reply:
OK so I hope your question is answered.